noir-projects/aztec-nr/aztec/src/context/inputs/public_context_inputs.nr

```diff
@@ -2,14 +2,15 @@ use dep::protocol_types::traits::Empty;
 
 // These inputs will likely go away once the AVM processes 1 public kernel per enqueued call.
 struct PublicContextInputs {
-    args_hash: Field,
+    // TODO: Remove this structure and get calldata size at compile time.
+    calldata_length: Field,
```

Comment:
I thought we were making calldata size a runtime variable?

Reply:
Yes yes, the difference is a bit subtle. The opcode takes an address (so, a runtime variable)

* We need to know the size at compile time for different reasons: to be able to create the [Field; N] where it will be stored, and then be able to pass that to the hasher.

Reply:
Okay, but I'm a bit confused. How can the size be known at compile time if it's a runtime variable?

Reply:
It's not known at compile time now, that's why the current code is different from the expected code (in the description).